src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -3135,6 +3135,58 @@ inline void compute_color_temperature_24_host(__m256 *p, __m256 pAdj)
     p[2] = _mm256_sub_ps(p[2], pAdj);    // color_temperature adjustment Bs
 }
 
+
+inline void compute_fog_48_host(__m256 *p, __m256 *pFogAlphaMask, __m256 *pFogIntensityMask)
+{
+    p[0] = _mm256_fmadd_ps(p[0],  _mm256_sub_ps(_mm256_set1_ps(1.0f), pFogAlphaMask[0]), _mm256_mul_ps(pFogIntensityMask[0], pFogAlphaMask[0]));    // fog adjustment
+    p[1] = _mm256_fmadd_ps(p[1],  _mm256_sub_ps(_mm256_set1_ps(1.0f), pFogAlphaMask[1]), _mm256_mul_ps(pFogIntensityMask[1], pFogAlphaMask[1]));    // fog adjustment
+    p[2] = _mm256_fmadd_ps(p[2],  _mm256_sub_ps(_mm256_set1_ps(1.0f), pFogAlphaMask[0]), _mm256_mul_ps(pFogIntensityMask[0], pFogAlphaMask[0]));    // fog adjustment
```

Comment:
It can load (R0 - R7) in p[0], (R8-R15) in p[1] ,  (G[0]- G[7]) in p[2], (G[8]-G[15]) in p[3], (B[0] -B[8] ) in p[4] and (B[8]- B[15]) in p[5] .
All three channels use same alpha and intensity value .
so p[0],p[2],p[4] use alpha[0] and p[1],p[3],p[5] use alpha[1]